### PR TITLE
Fix gsd.hoomd.Frame not supported 

### LINKED ISSUE #1131: `NeighborQuery` does not recognize `gsd.hoomd.Frame`
### Description

`NeighborQuery` casts from `gsd.hoomd.Snapshot`, but this class was renamed `gsd.hoomd.Frame` in GSD 3.

The fix is to add `'gsd.hoomd.Frame'` to this list of duck-type snapshots:
https://github.com/glotzerlab/freud/blob/965b0fa0c52d3eded1f1fae7e94c0526d6dae0fc/freud/locality.pyx#L325-L328

### Steps to Reproduce

```python
import freud
import gsd.hoomd

rdf = freud.density.RDF(bins=60, r_max=3.0)
with gsd.hoomd.open("trajectory.gsd") as traj:
    for snap in traj:
        rdf.compute(snap, reset=False)
rdf.plot()
```

### Error Output

```shell
TypeError                                 Traceback (most recent call last)
Cell In[10], line 7
      5 with gsd.hoomd.open(result.directory.file("trajectory.gsd")) as traj:
      6     for snap in traj:
----> 7         rdf.compute(snap,reset=False)
      8 rdf.plot();

File util.pyx:165, in freud.util._Compute.__getattribute__.compute_wrapper()

File density.pyx:660, in freud.density.RDF.compute()

File locality.pyx:992, in freud.locality._PairCompute._preprocess_arguments()

File locality.pyx:375, in freud.locality.NeighborQuery.from_system()

TypeError: 'Frame' object is not iterable

Error: Process completed with exit code 1.
```

### freud Version

2.13.0

### Python Version

3.8

### System Platform

Linux

### Installation method

Download from PyPI

### Developer

None

### PR DESCRIPTION
**Pull request by DomFijan**
Recent change in gsd3.0 deprecated use of gsd.hoomd.Snapshot with gsd.hoomd.Frame. This should fix that.
 
## Description
Freud does not support use of gsd.hoomd.Frame yet.

## Motivation and Context
Resolves: #1131 

## How Has This Been Tested?
I added some tests that use NeighborQuery.from_system() to construct a freud neighborlist object.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

**Comment by DomFijan** [comment]
Test are failing partially due to the fact garnett is incompatible with gsd3.0 if I'm reading this correctly.

File: tests/test_locality_NeighborQuery.py [hunk 1]
```diff
@@ -1,19 +1,29 @@
 # Copyright (c) 2010-2023 The Regents of the University of Michigan
 # This file is from the freud project, released under the BSD 3-Clause License.
 
+import copy as cp
```
Comment (bdice) [inline tests/test_locality_NeighborQuery.py]: Sometimes `cp` is used to refer to `cupy` (mirroring `numpy as np`). Let's be explicit and use the built-in name.
```suggestion
import copy
```

File: tests/test_locality_NeighborQuery.py [hunk 2]
```diff
@@ -703,3 +713,35 @@ def test_alternating_points(self):
...
+        snap.particles.N = 10
+        snap.particles.position = np.random.rand(10, 3) * 10
+        snap.configuration.box = [10, 10, 10, 0, 0, 0]
+        list_systems.append(("gsd-snapshot", snap))
+    return list_systems
+
+
+@pytest.fixture(scope="module", params=_from_system_inputs(), ids=(lambda x: x[0]))
+def system(request):
+    return cp.deepcopy(request.param[1])
```
Comment (bdice) [inline tests/test_locality_NeighborQuery.py]:
```suggestion
    return copy.deepcopy(request.param[1])
```

**Review by joaander — APPROVED** [review]
Thanks!

Will #1133 update `requirements-test.txt` to test with`gsd==3.0.1`?

**Comment by tommy-waltmann** [comment]
> Thanks!
> 
> Will #1133 update `requirements-test.txt` to test with`gsd==3.0.1`?

I think the incoming landslide of dependabot PRs will do that :)

**Pull request merged**